java/src/jmri/jmrix/can/cbus/CbusSensorManager.java

```diff
@@ -100,24 +100,27 @@ public String getNextValidAddress(String curAddress, String prefix) throws JmriE
         } catch (IllegalArgumentException e) {
             throw new JmriException(e.toString());
         }
-        // If the hardware address passed does not already exist then this can
-        // be considered the next valid address.
+        testAddr = CbusAddress.validateSysName(testAddr); // normalize Merg address
         Sensor s = getBySystemName(prefix + typeLetter() + testAddr);
-        if (s == null) {
-            return testAddr;
-        }
-        // getIncrement will have performed a max check on the numbers
-        String newaddr = CbusAddress.getIncrement(testAddr);
-        if (newaddr == null) {
+        if (s != null) {
+            // build local addresses
+            for (int x = 1; x < 10; x++) {
```

Comment:
@icklesteve : Cbus Sensors (and Turnouts, not Lights) now look ahead 10 addresses for an available one, like Lenz, Loconet managers.